### DISCUSSION TITLE
Exporting a subset of the documentation

### DISCUSSION THREAD

**Discussion by f-ricci** [facebook/docusaurus#9270] [Q&A] [answered]
Hi, we are using docusaurus to document every API we have internally and we would like to share some of that documentation with other developers but not give access to everything.

Is there any way to export or build only a subset of the documentation?

Thanks for the help!

**Comment by slorber** [comment, marked as answer]
We don't have a first-class feature for this, but it is possible. 

You can build the same site into multiple variants, one of them exposing less information. 

For example
- You can remove docs from the `./docs` folder as part of a prebuild step. It can be as simple as running `rm -rf docs/internal`.
- You can use an env variable to define different site config, docs sidebars depending on which variant you are currently building.
- You can create a Remark plugin and invent your own markdown syntax that show/hide some documentation paragraphs that are only relevant to a given variant.

Facebook does this internally for its own open-source projects docs: they build one public variant and they also build it for internal usage, including more pages and extra content.

----

Some Facebook internal/external example refs:
- https://github.com/facebook/relay/blob/main/website/sidebars.js
- https://www.npmjs.com/package/docusaurus-plugin-internaldocs-fb?activeTab=code

If you study a bit how it works under the hood you'll see that they have helper functions that return an internal or external config or sidebar slice depending on the `process.env.FB_INTERNAL` variable.

**Reply by f-ricci** [reply]
I will give it a try.

Thanks!
